addons/mail/models/mail_thread.py

```diff
@@ -2041,6 +2041,10 @@ def message_post_with_template(self, template_id, email_layout_xmlid=None, auto_
         res_id = kwargs.get('res_id', self.ids and self.ids[0] or 0)
         res_ids = kwargs.get('res_id') and [kwargs['res_id']] or self.ids
 
+        template = self.env['mail.template'].browse(template_id).exists()
```

Comment:
That's not the purpose of this method to check the template existence, as it adds queries for every calls. Instead it is caller's job to ensure the API is called with the right parameters, and trying to find a fallback if the template does not exists.